app/assets/javascripts/components/common/calendar.jsx

```diff
@@ -34,12 +42,6 @@ const Calendar = createReactClass({
     }
     return true;
   },
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    if (nextProps.course.start !== moment(this.state.initialMonth).format('YYYY-MM-DD')) {
```

Comment:
is this condition not needed?

Reply:
I talked to @ragesoss in Slack about this. Firstly the statement is being compared in a wrong way, i.e., ISO String to yyyy-mm-dd (ISO String contains both date and time). Secondly, when I corrected this comparison and shifted it to `getDerivedStateFromProps`, @ragesoss told me to leave this statement as we were calling `moment` twice in this and this is not really necessary and to leave this condition altogether.